### PR TITLE
Cleanup and improve unit test coverage for transformPointCloud functions

### PR DESCRIPTION
**Pull request by taketwo**

### CODE REVIEW

File: test/common/test_centroid.cpp [hunk 1]
```diff
@@ -1009,9 +1012,93 @@ TEST (PCL, computeCentroid)
...
+  demeanPointCloud (cloud, centroid, cloud_demean);
+  EXPECT_EQ (cloud_demean.is_dense, cloud.is_dense);
+  EXPECT_EQ (cloud_demean.width, cloud.width);
+  EXPECT_EQ (cloud_demean.height, cloud.height);
+  EXPECT_EQ (cloud_demean.size (), cloud.size ());
+
+  EXPECT_NEAR (cloud_demean[0].x, 0.034503, 1e-4);
+  EXPECT_NEAR (cloud_demean[0].y, 0.010837, 1e-4);
+  EXPECT_NEAR (cloud_demean[0].z, 0.013447, 1e-4);
+
```
[the remark this answers is not in the document]
Comment (taketwo) [reply test/common/test_centroid.cpp]: Firstly, there is no `.w` member, only `data[3]`. Secondly, is it the part of the contract that this padding number is always 1.0? I'm not sure I ever seen this stated explicitly anywhere.

[the remark this answers is not in the document]
Comment (SergioRAgostinho) [reply test/common/test_centroid.cpp]: > Secondly, is it the part of the contract that this padding number is always 1.0?

I makes sense to initialize by default with 1.0, but from then onwards, it's we can't assume anything anymore.

I'm not really sure how to handle this. In a sense, in most cases we're likely to have the homogeneous coordinate set to the same value in all points. But if it's not, I don't think demeanPointCloud is even handling it. So yeah, you've got a point.

**Comment by taketwo** [comment]
Thanks for taking time to review. I pushed one small commit addressing one of the points you made. I fully agree with all other, but I don't have cycles now to clean up messy unit tests :(

(I guess you know where I am heading with this PR. To be honest my "ok, let's just try to use Eigen multiplication for transforms" project took way more effort that I anticipated.)

**Comment by SergioRAgostinho** [comment]
> (I guess you know where I am heading with this PR. To be honest my "ok, let's just try to use Eigen multiplication for transforms" project took way more effort than I anticipated.)

Classic :) I believe anything less of what you made you leave room for doubt and questioning. Now we're able to quickly benchmark multiple configurations and settings and easily interpret the results. Also, if you think about it, you just wrote a boilerplate for whatever benchmark you might need to do in the future. To be honest, for my future ones as well ^^ It should payoff in the long run.

**Pull request merged**